From a534dff540b9183a30d524967ccdc5d087819165 Mon Sep 17 00:00:00 2001
Message-Id: <a534dff540b9183a30d524967ccdc5d087819165.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Wed, 1 Oct 2014 17:18:10 +0530
Subject: [PATCH 303/312] FOR_UPSTREAM [VPG]: drm/i915: Fix the linear offset
 for rotation.

Patch fixes the linear offset for rotation.

Issue: GMINL-1872
Change-Id: Ib3395cd12493abb2fcb2101e9e463dbdc7cdf2d3
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 8ed127a..c81a865 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2697,7 +2697,7 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 			I915_WRITE(DSPLINOFF(plane),
 				   linear_offset +
 				   (fb->height - 1) * fb->pitches[0] +
-				   fb->width * pixel_size);
+				   (fb->width - 1) * pixel_size);
 		} else {
 			I915_WRITE(DSPTILEOFF(plane), (y << 16) | x);
 			I915_WRITE(DSPLINOFF(plane), linear_offset);
-- 
1.7.9.5

